Update debug code to not use Startup /2 (#35339)
dotnet/AspNetCore.Docs: aspnetcore/test/troubleshoot.md

> To obtain the environment, use either of the following approaches:
> 

Reviewer: The 9,x code shows only one approach

```suggestion

```

> * Assign the environment to a property in `Program.cs`. Check the environment using the property, for example, `if (app.Environment.IsDevelopment())`.
> * Inject `IHostingEnvironment` and check the environment with the local variable. The following sample code demonstrates this approach.

Reviewer: ```suggestion
Get the environment from the `Environment` property of `WebApplication`. For example, `if (app.Environment.IsDevelopment())` as in the following sample code.
```
